src/Illuminate/Validation/Rules/IsInstanceOf.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Illuminate\Validation\Rules;
+
+use Illuminate\Contracts\Validation\Rule;
+use Illuminate\Contracts\Validation\ValidatorAwareRule;
+use Illuminate\Support\Traits\Conditionable;
+use TypeError;
+
+class IsInstanceOf implements Rule, ValidatorAwareRule
+{
+    use Conditionable;
+
+    /**
+     * The class of the object to validate against.
+     *
+     * @var class-string
+     */
+    protected $object;
+
+    /**
+     * Whether to check each item of an iterable.
+     *
+     * @var bool
+     */
+    protected $isIterable = false;
+
+    /**
+     * Whether to enforce strict checking of the object type
+     * meaning that null values will fail the validation.
+     *
+     * @var bool
+     */
+    protected $strict = false;
+
+    /**
+     * The current validator instance.
+     *
+     * @var \Illuminate\Validation\Validator
+     */
+    protected $validator;
+
+    /**
+     * Create a new rule instance.
+     *
+     * @param  class-string  $object
+     */
+    public function __construct($object)
+    {
+        if (! is_string($object)) {
+            throw new TypeError('The object must be a class string.');
+        }
```

Comment:
Strictly speaking, the error message does not match the condition. It's never checked if the given string actually is a _class_ string